### ISSUE TITLE
cuDNN launch failure for Resnet Encoders

### ISSUE THREAD

**Issue by bohaohuang** [qubvel/segmentation_models#44]
I have the following error message when running segmentation model with resnet as encoder:
InternalError (see above for traceback): cuDNN launch failure : input shape ([5,3,576,576])
	 [[node bn_data/FusedBatchNorm (defined at /home/lab/anaconda3/envs/tf-aml/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:1806)  = FusedBatchNorm[T=DT_FLOAT, _class=["loc:@training/Adam/gradients/AddN_249"], data_format="NCHW", epsilon=2e-05, is_training=true, _device="/job:localhost/replica:0/task:0/device:GPU:0"](training/Adam/gradients/bn_data/FusedBatchNorm_grad/FusedBatchNormGrad-1-TransposeNHWCToNCHW-LayoutOptimizer, bn_data/Const_3, bn_data/beta/read, decoder_stage4_bn2/Const_4, decoder_stage4_bn2/Const_4)]]
	 [[{{node loss/mul/_3415}} = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_22986_loss/mul", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]

It seems this is related with cuDNN and fused batch norm. I've searched on the internet about the related topics and updated my drivers but still cannot resolve this. My system information is shown below:
GPU: RTX 2080Ti
Driver: 410.57
CUDA: 9.0
cuDNN: 7.4.2
Keras: 2.1.6 (I down-grade to this version because the newer version seems causing problems for batch norm)
Tensorflow: 1.12

**Comment by qubvel** [comment]
Hi @bohaohuang,
All test have passed for tensorflow 1.9, can you try this one?

**Comment by bohaohuang** [comment]
I tried to run it in a tf 1.9 version with other settings unchanged:
GPU: RTX 2080Ti
Driver: 410.57
CUDA: 9.0
cuDNN: 7.4.2
Keras: 2.1.6
Tensorflow: 1.9

And still have the same problem.

**Comment by qubvel** [comment]
Works with the following configuration:
GPU 1080Ti
Driver 390.48
CUDA 9.0
Keras 2.1.6
Tensorflow 1.9

Did you try any other networks? for example, smth from keras_applications just to check?

**Comment by bohaohuang** [comment]
I used a resnet50 model in keras (https://keras.io/applications/#resnet50) and it worked fine.
I have a 1080ti actually, I'm going to try on that one next and use your configuration.

**Comment by qubvel** [comment]
I assume that's not the problem of segmentation_models but tensorflow/cuda/drivers compatibility

**Comment by bohaohuang** [comment]
Ok, I have tried your settings on a 1080ti machine. It works without having any problem at all. Thanks for your advice, I think it might be the cuda/cudnn issues with the new cards. 
I'll try to see if the same problem happens when I use fused batch norm in keras with 2080s. I might start a new thread there.

**Comment by eram1205** [comment]
@bohaohuang I have the same problem. Could you solve it? Thanks!

**Comment by bohaohuang** [comment]
> @bohaohuang I have the same problem. Could you solve it? Thanks!

I updated tf and cuda and never had this issue again. Maybe you can try that if you're not using CUDA 10.1

**Comment by eram1205** [comment]
> > @bohaohuang I have the same problem. Could you solve it? Thanks!
> 
> I updated tf and cuda and never had this issue again. Maybe you can try that if you're not using CUDA 10.1

Thanks a lot!
